app/scripts/metamask-controller.js

```diff
@@ -521,6 +547,72 @@ export default class MetamaskController extends EventEmitter {
       subjectCacheLimit: 100,
     });
 
+    ///: BEGIN:ONLY_INCLUDE_IN(flask)
+    this.workerController = new IframeExecutionEnvironmentService({
+      onError: this.onExecutionEnvironmentError.bind(this),
+      iframeUrl: new URL(
+        'https://metamask.github.io/iframe-execution-environment/0.3.1',
+      ),
+      messenger: this.controllerMessenger.getRestricted({
+        name: 'ServiceMessenger',
+      }),
+      setupSnapProvider: this.setupSnapProvider.bind(this),
+    });
+
+    const snapControllerMessenger = this.controllerMessenger.getRestricted({
+      name: 'SnapController',
+      allowedEvents: [
+        'ServiceMessenger:unhandledError',
+        'ServiceMessenger:unresponsive',
+      ],
+      allowedActions: [
+        `${this.permissionController.name}:hasPermission`,
+        `${this.permissionController.name}:getEndowments`,
+      ],
+    });
+
+    this.snapController = new SnapController({
+      endowmentPermissionNames: Object.values(EndowmentPermissions),
+      terminateAllSnaps: this.workerController.terminateAllSnaps.bind(
+        this.workerController,
+      ),
+      terminateSnap: this.workerController.terminateSnap.bind(
+        this.workerController,
+      ),
+      executeSnap: this.workerController.executeSnap.bind(
+        this.workerController,
+      ),
+      getRpcMessageHandler: this.workerController.getRpcMessageHandler.bind(
+        this.workerController,
+      ),
+      // TODO:flask Fix in skunkworks (SnapController)
```

Comment:
These two TODO comments, what needs to be fixed for them exactly?

Reply:
Using controller messenger actions instead of passing around functions. This has been done as of `skunkworks@0.8.1`.